src/operations/query/r4.js

```diff
@@ -17,6 +17,129 @@ const {searchParameterQueries} = require('../../searchParameters/searchParameter
+                        queryParameterValue = queryParameterValue['values'];
+                    }
+                    break;
+                case 'token':
+                    if (queryParameterValue['value']) {
+                        queryParameterValue['values'] = [queryParameterValue['value']];
+                    }
+                    if (queryParameterValue['values']) {
+                        for (const token of queryParameterValue['values']) {
+                            queryParameterValue = [];
+                            let tokenString = '';
+                            if (token['system']) {
+                                tokenString = token['system'] + '|';
+                            }
+                            if (token['code']) {
+                                tokenString += token['code'];
+                            }
+                            if (token['value']) {
+                                tokenString += token['value'];
+                            }
+                            if (tokenString) {
+                                queryParameterValue.push(tokenString);
+                            }
+                        }
+                    }
+                    break;
+                case 'reference':
+                    // eslint-disable-next-line no-case-declarations
+                    let referenceText = '';
+                    if (queryParameterValue['target']) {
+                        referenceText = queryParameterValue['target'] + '/';
+                    }
+                    if (queryParameterValue['value']) {
+                        referenceText += queryParameterValue['value'];
+                    }
+                    queryParameterValue = referenceText;
+                    break;
+                case 'quantity':
+                    // eslint-disable-next-line no-case-declarations
+                    let quantityString = '';
+                    if (queryParameterValue['prefix']) {
+                        quantityString += queryParameterValue['prefix'];
+                    }
+                    if (queryParameterValue['value']) {
+                        quantityString += queryParameterValue['value'];
+                    }
+                    if (queryParameterValue['system']) {
+                        quantityString = '|' + queryParameterValue['system'];
+                    }
+                    if (queryParameterValue['code']) {
+                        quantityString = '|' + queryParameterValue['code'];
+                    }
+                    queryParameterValue = quantityString;
+                    break;
+                case 'date':
+                case 'dateTime':
+                case 'number':
+                    if (queryParameterValue['value']) {
+                        queryParameterValue['values'] = [queryParameterValue['value']];
+                    }
+                    if (queryParameterValue['values']) {
```

Comment:
Should this be its own module that could be reused for an STU3/R5 query operations?

Reply:
yeah, eventually.  but probably in a separate PR.